src/analyses/base.ml

```diff
@@ -2732,11 +2756,7 @@ struct
     | Malloc size, _ -> begin
         match lv with
         | Some lv ->
-          let heap_var =
-            if (get_bool "sem.malloc.fail")
-            then AD.join (AD.of_var (heap_var false man)) AD.null_ptr
-            else AD.of_var (heap_var false man)
-          in
+          let heap_var = alloc Queries.AllocationLocation.Heap size in
```

Comment:
I now realized something else about this allocation code: even in the case where this _only_ returns a `NULL` pointer, we put a blob into the local state.
But this blob isn't pointed at by anything, so it just clutters the local state.

Ideally we wouldn't have such an unreachable blob, but this can only happen under rare conditions, so it should barely matter. But it might also be worth commenting about.

Reply:
I don't think that's true after these changes though? The call to `alloced_var` is guarded by `include_pointer`.

Reply:
> Then we're writing into NULL but that isn't producing any warning?

I'm not sure if it is the `set` that should produce the warning here either way?